test/Microsoft.AspNet.Server.KestrelTests/PathNormalizerTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using Microsoft.AspNet.Server.Kestrel.Http;
+using Xunit;
+
+namespace Microsoft.AspNet.Server.KestrelTests
+{
+    public class PathNormalizerTests
+    {
+        [Theory]
+        [InlineData("/a", "/a")]
+        [InlineData("/a/", "/a/")]
+        [InlineData("/a/b", "/a/b")]
+        [InlineData("/a/b/", "/a/b/")]
+        [InlineData("", ".")]
```

Comment:
Path should always start with `/`, so `.` and `..` are undefined, you only need to check for `/.`, `/./`, `/..`, and `/../`